File: solutions/338/338.swift
class Solution {
    func countBits(_ n: Int) -> [Int] {
        var ans = [Int]()
        
        for i in 0...n {
            var number = i
            var sum = 0

            while number != 0 {
                sum += number % 2
                number /= 2
            }
            
            ans.append(sum)
        }

        return ans
    }
}

File: solutions/338/338.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> ans;
        
        for(int i = 0; i <= n; i++) {
            int number = i;
            int sum = 0;

            while(number != 0) {
                sum += number % 2;
                number = number / 2;
            }
            
            ans.push_back(sum);
        }

        return ans;
    }
};